From 4cf9e15277dba6c4b7dcbb6c1d10e5b7fd4d8ac0 Mon Sep 17 00:00:00 2001
From: Karl Hobley <karl@torchbox.com>
Date: Wed, 13 Oct 2021 15:58:17 +0100
Subject: [PATCH] Fix core/edit_handlers relative imports

---
 wagtail/core/edit_handlers.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/wagtail/core/edit_handlers.py b/wagtail/core/edit_handlers.py
index ae7deeec9c..62752e8777 100644
--- a/wagtail/core/edit_handlers.py
+++ b/wagtail/core/edit_handlers.py
@@ -29,9 +29,9 @@ from wagtail.utils.decorators import cached_classmethod
 # DIRECT_FORM_FIELD_OVERRIDES, FORM_FIELD_OVERRIDES are imported for backwards
 # compatibility, as people are likely importing them from here and then
 # appending their own overrides
-from .forms.models import (  # NOQA
+from .admin.forms.models import (  # NOQA
     DIRECT_FORM_FIELD_OVERRIDES, FORM_FIELD_OVERRIDES, WagtailAdminModelForm, formfield_for_dbfield)
-from .forms.pages import WagtailAdminPageForm
+from .admin.forms.pages import WagtailAdminPageForm
 
 
 def widget_with_script(widget, script):
-- 
2.27.0

