From 08656cfd94d15308a3667b14541167b6be05e259 Mon Sep 17 00:00:00 2001
From: Karl Hobley <karl@kaed.uk>
Date: Mon, 21 Mar 2022 13:55:47 +0000
Subject: [PATCH] Fixup commenting models

---
 wagtail/models/commenting.py | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/wagtail/models/commenting.py b/wagtail/models/commenting.py
index e1fa03e62e..a555ed7869 100644
--- a/wagtail/models/commenting.py
+++ b/wagtail/models/commenting.py
@@ -1,6 +1,5 @@
 from django.conf import settings
 from django.db import models
-from wagtail.models import Page, COMMENTS_RELATION_NAME, PageRevision
 from django.utils.translation import gettext_lazy as _
 from modelcluster.fields import ParentalKey
 from wagtail.log_actions import log
@@ -18,7 +17,7 @@ class Comment(ClusterableModel):
     """
 
     page = ParentalKey(
-        Page, on_delete=models.CASCADE, related_name=COMMENTS_RELATION_NAME
+        'Page', on_delete=models.CASCADE, related_name=COMMENTS_RELATION_NAME
     )
     user = models.ForeignKey(
         settings.AUTH_USER_MODEL,
@@ -38,7 +37,7 @@ class Comment(ClusterableModel):
     updated_at = models.DateTimeField(auto_now=True)
 
     revision_created = models.ForeignKey(
-        PageRevision,
+        'PageRevision',
         on_delete=models.CASCADE,
         related_name="created_comments",
         null=True,
@@ -168,7 +167,7 @@ class PageSubscription(models.Model):
         on_delete=models.CASCADE,
         related_name="page_subscriptions",
     )
-    page = models.ForeignKey(Page, on_delete=models.CASCADE, related_name="subscribers")
+    page = models.ForeignKey('Page', on_delete=models.CASCADE, related_name="subscribers")
 
     comment_notifications = models.BooleanField()
 
-- 
2.35.1

