From 0a73fa0c4dc0e91c905dfd2a11cfc8cabc2978ab Mon Sep 17 00:00:00 2001
From: Karl Hobley <karl@torchbox.com>
Date: Thu, 17 Feb 2022 16:35:09 +0000
Subject: [PATCH] Move admin static to top level app

---
 client/webpack.config.js | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/client/webpack.config.js b/client/webpack.config.js
index 18a9bc943..7b23b2c74 100644
--- a/client/webpack.config.js
+++ b/client/webpack.config.js
@@ -12,6 +12,10 @@ const getOutputPath = (app, folder, filename) => {
     'contrib/modeladmin': 'wagtailmodeladmin',
   };
 
+  if (app === 'admin') {
+    return path.join('wagtail', 'static', 'wagtailadmin', folder, filename);
+  }
+
   const appLabel = exceptions[app] || `wagtail${app}`;
 
   return path.join('wagtail', app, 'static', appLabel, folder, filename);
-- 
2.30.2

