From 3999b3c198e70e4df788247cbb39b3062372725d Mon Sep 17 00:00:00 2001
From: Karl Hobley <karl@kaed.uk>
Date: Mon, 21 Mar 2022 14:21:26 +0000
Subject: [PATCH] Pages models: Fixup imports

---
 wagtail/core/models/__init__.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/wagtail/core/models/__init__.py b/wagtail/core/models/__init__.py
index fdbbf1c63f..894453ba58 100644
--- a/wagtail/core/models/__init__.py
+++ b/wagtail/core/models/__init__.py
@@ -1,4 +1,4 @@
-from wagtail.models import (  # noqa
+from wagtail.models.pages import (  # noqa
     PAGE_MODEL_CLASSES,
     PAGE_PERMISSION_TYPE_CHOICES,
     PAGE_PERMISSION_TYPES,
-- 
2.35.1

